api/src/main/java/uk/gov/dhsc/htbhf/claimant/repository/ClaimantRepository.java

```diff
@@ -5,17 +5,45 @@
 import org.springframework.data.repository.query.Param;
 import uk.gov.dhsc.htbhf.claimant.entity.Claimant;
 
+import java.util.Optional;
 import java.util.UUID;
 
 /**
  * JPA repository for the Claimant table.
  */
 public interface ClaimantRepository extends CrudRepository<Claimant, UUID> {
 
-    @Query("SELECT COUNT(claimant) FROM Claimant claimant WHERE claimant.nino = :nino AND claimant.eligibilityStatus = 'ELIGIBLE'")
-    Long getNumberOfMatchingEligibleClaimants(@Param("nino") String nino);
+    @Query("SELECT COUNT(claimant) "
+            + "FROM Claimant claimant "
+            + "WHERE claimant.nino = :nino "
+            + "AND claimant.eligibilityStatus = 'ELIGIBLE'")
+    Long getNumberOfMatchingEligibleClaimantsWithNino(@Param("nino") String nino);
 
-    default Boolean eligibleClaimExists(String nino) {
-        return getNumberOfMatchingEligibleClaimants(nino) != 0;
+    @Query("SELECT COUNT(claimant) "
+            + "FROM Claimant claimant "
+            + "WHERE claimant.dwpHouseholdIdentifier = :dwpHouseholdIdentifier "
+            + "AND claimant.eligibilityStatus = 'ELIGIBLE'")
+    Long getNumberOfMatchingEligibleClaimantsWithDwpHouseholdIdentifier(@Param("dwpHouseholdIdentifier") String dwpHouseholdIdentifier);
```

Comment:
how about 'countEligibleClaimantsWithDwpHouseholdIdentifier'? We'd probably want to change all 'getNumberOf' methods if we adopt this name instead.